components/data_prefetch/data_prefetch.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PQueue from 'p-queue';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {Constants} from 'utils/constants';
+import {loadProfilesForSidebar} from 'actions/user_actions.jsx';
+
+const queue = new PQueue({concurrency: 2});
+
+type Props = {
+    currentChannelId: string;
+    prefetchQueueObj: Record<string, string[]>;
+    prefetchRequestStatus: Record<string, any>;
+    unreadChannels: Channel[];
+    actions: {
+        prefetchChannelPosts: (channelId: string, delay: number | undefined) => Promise<any>;
+    };
+}
+
+export default class DataPrefetch extends React.PureComponent<Props, {}> {
```

Comment:
I don't feel this is a good example of usage for functional components or hooks. This is not a reusable component or needs any abstraction for the state. Both of them work great when we have multiple of these component instances in the app like dot menu etc where the benefit of a smaller footprint component is magnified because we can have thousands of them at the cost of small optimization for react rendering. 

Here, in this case, I don't think it fits the bill in either purpose

Reply:
I cross-checked this again but it does not help much in this case. we could however do this for other components in the future mainly when we update redux to 7.1v to use `useSelector` or `useDispatch` etc otherwise here we will end up with one component for connecting to redux and then another hook component making this more complex than it needs to be.